custom_components/tuya_local/devices/poolsana_heatpump.yaml

```diff
@@ -0,0 +1,85 @@
+name: Pool heat pump
+primary_entity:
+  products:
+  - id: abb8kaxcphfzkwsg
+    name: PoolSana InverPower Next
+  entity: climate
+  translation_key: pool_heatpump
+  dps:
+    - id: 1
+      type: boolean
+      name: hvac_mode
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          constraint: mode
+          conditions:
+            - dps_val: Cooling
+              value: cool
+            - dps_val: Heating
+              value: heat
+            - dps_val: Auto
+              value: auto
+    - id: 2
+      type: string
+      name: mode
+      hidden: true
+    - id: 4
+      type: integer
+      name: temperature
+      range:
+        min: 70
+        max: 400
+      mapping:
+        - scale: 10
+          step: 10
+    - id: 16
+      type: integer
+      name: current_temperature
+      range:
+        min: -300
+        max: 1000
+      mapping:
+        - scale: 10
+          step: 1
+    - id: 5
+      type: string
+      name: preset_mode
+      mapping:
+        - dps_val: Boost
+          value: boost
+        - dps_val: Silence
+          value: eco
+        - dps_val: Smart
+          value: comfort
+secondary_entities:
+  - entity: sensor
+    class: temperature
+    name: Effluent Temperature
+    category: diagnostic
+    dps:
+      - id: 25
+        type: integer
+        name: sensor
+        range:
```

Comment:
Again, range is not needed for sensors, as they report what the device is saying regardless of whether it is in range or not. Step and scale are also both at defaults, so the whole mapping can be removed too.